### PR TITLE
Adding Autotaunt Timer/UI

### PR DESCRIPTION
**Pull request by bsnuggs1**
I was finally able to get around to separating most of the autotaunt code and the UI associated with it.  The only changes in the core files I couldn't avoid were the hooks being made to control taunt/autotaunting variables.  If you have any suggestions, I can move that code into a separate file as well.

I also added some more taunts, but I can remove those from the request if you like.

### CODE REVIEW

**Comment by Newbrict** [comment]
I can't read through this right now but I'll test it tonight and hopefully merge it :*)

Thanks.

File: gamemode/server/rounds.lua [hunk 1]
```diff
@@ -89,21 +89,21 @@ local function InRound()
...
+		 return
+	 end
+
+	 if( #hunters == 0 ) then
+		 round.state = ROUND_END
+		 round.endTime = CurTime()
+		 round.winner = "Props"
+		 hook.Call( "OBJHUNT_RoundEnd" )
+		 return
+	 end
```
Comment (Newbrict) [inline gamemode/server/rounds.lua]: there's no reason to add a space before each line here

File: gamemode/shared/sh_config.lua [hunk 2]
```diff
@@ -130,26 +136,41 @@ TAUNT_MIN_PITCH = 64
...
 
 HUNTER_TAUNTS = {}
 -- USAGE:
 -- HUNTER_TAUNTS["Display Name"] = "taunts/file_name.wav"
-HUNTER_TAUNTS["Jaws Theme"]       = "taunts/jaws.wav"
-HUNTER_TAUNTS["I Will Find You"]  = "taunts/taken.wav"
-HUNTER_TAUNTS["Where Are You!"]   = "taunts/where.wav"
+HUNTER_TAUNTS["GlaDoS - President"]     = "taunts/glados-president.wav"
+HUNTER_TAUNTS["Come To Papa"]     		= "taunts/come_to_papa.wav"
+HUNTER_TAUNTS["No Soul"]   		 		= "taunts/soul.wav"
```
Comment (Newbrict) [inline gamemode/shared/sh_config.lua]: spacing here is messed up

Comment (bsnuggs1) [inline gamemode/shared/sh_config.lua]: I will fix that up and re-commit.

File: gamemode/init.lua [hunk 3]
```diff
@@ -376,7 +401,7 @@ net.Receive( "Prop Angle Lock", function( len, ply )
 	net.Start( "Prop Angle Lock BROADCAST" )
 		net.WriteEntity( ply )
 		net.WriteBit( lockStatus )
-		net.WriteAngle( propAngle )
+		net.WriteAngle( propAngle) 
 	net.Broadcast()
```
Comment (Newbrict) [inline gamemode/init.lua]: add the space back

File: gamemode/init.lua [hunk 4]
```diff
@@ -215,12 +241,11 @@ hook.Add( "Initialize", "Precache all network strings", function()
 	util.AddNetworkString( "Prop Angle Lock BROADCAST" )
 	util.AddNetworkString( "Prop Angle Snap" )
 	util.AddNetworkString( "Prop Angle Snap BROADCAST" )
+    util.AddNetworkString( "AutoTaunt Update" )
+    util.AddNetworkString( "Update Taunt Times" )
```
Comment (Newbrict) [inline gamemode/init.lua]: spacing incorrect for these lines

File: gamemode/server/rounds.lua (hunk 1 above)
Comment (bsnuggs1) [inline gamemode/server/rounds.lua]: That's odd, I am wondering if the carriage returns and line feeds were normalized when I was using visual studio.  Nevertheless, I will take care of this.

File: gamemode/init.lua [hunk 5]
```diff
@@ -106,6 +131,7 @@ function GM:PlayerSetModel( ply )
 		-- default
 		ply:SetViewOffset( Vector(0,0,64) )
 	elseif( class == "player_prop" ) then
+        print( "Setting as Prop" )
```
Comment (Newbrict) [inline gamemode/init.lua]: spacing incorrect

File: gamemode/cl_init.lua [hunk 6]
```diff
@@ -18,6 +18,7 @@ net.Receive( "Prop update", function( length )
 	local propHeight = tHitboxMax.z - tHitboxMin.z
 	LocalPlayer().propHeight = propHeight
 	LocalPlayer().lastPropChange = CurTime()
+    LocalPlayer().lastTaunt = CurTime()
```
Comment (Newbrict) [inline gamemode/cl_init.lua]: spacing incorrect

File: gamemode/cl_init.lua [hunk 7]
```diff
@@ -37,9 +39,10 @@ end )
 net.Receive( "Reset Prop", function( length )
 	-- taunt default
 	LocalPlayer().nextTaunt = 0
-	LocalPlayer().lastTaunt = 0
+	LocalPlayer().lastTaunt = CurTime()
 	LocalPlayer().lastTauntDuration = 1
 	LocalPlayer().lastTauntPitch = 100
+    LocalPlayer().autoTauntInterval = OBJHUNT_AUTOTAUNT_INTERVAL + OBJHUNT_HIDE_TIME
```
Comment (Newbrict) [inline gamemode/cl_init.lua]: spacing incorrect

**Comment by SunRed** [comment]
Will this ever be merged or do I have to implement it myself?

**Comment by Newbrict** [comment]
I want it to be merged but the PR isn't great, there are a few formatting issues, SunRed can you add me on steam.

**Comment by Newbrict** [comment]
merged

**Pull request closed without merging**